src/MgvOfferTaking.sol

```diff
@@ -235,7 +188,8 @@ abstract contract MgvOfferTaking is MgvHasOffers {
           2. `sor.gives` was at most `mor.initialGives - mor.totalGave` from earlier step,
           3. `sor.gives` may have been clamped _down_ during `execute` (to "`offer.wants`" if the offer is entirely consumed, or to `makerWouldWant`, cf. code of `execute`).
         */
-        (sor.offerId, sor.local) = getNextBest(offerList, mor, sor.offer, sor.local, sor.olKey.tickScale);
+        (mor.leaf, sor.local) = dislodgeOffer(offerList, sor.olKey.tickScale, offer.prev(0), sor.local, true, mor.leaf);
+        sor.offerId = mor.leaf.getNextOfferId();
```

Comment:
I like how this makes it clearer that we're removing `offer` and then getting the next `offerId`.